From b60312ac8119f89cf093dacfea0e9f4ce6801818 Mon Sep 17 00:00:00 2001
From: Alisher Alikhodjaev <alisher@google.com>
Date: Wed, 1 Mar 2023 13:00:03 -0800
Subject: [PATCH] OOBR in AnalyzeMfcResp in NxpMfcReader.cc

Bug: 252763983
Test: build ok
Change-Id: I91fa035ca6245e6039eeedb447d7e3306b7aebc5
(cherry picked from commit on googleplex-android-review.googlesource.com host: 69c53a6f1bca6d450e5100c1044114ffad615e5c)
Merged-In: I91fa035ca6245e6039eeedb447d7e3306b7aebc5
---
 pn8x/halimpl/mifare/NxpMfcReader.cc  | 4 ++++
 snxxx/halimpl/mifare/NxpMfcReader.cc | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/pn8x/halimpl/mifare/NxpMfcReader.cc b/pn8x/halimpl/mifare/NxpMfcReader.cc
index 602a2b7..9ee49db 100644
--- a/pn8x/halimpl/mifare/NxpMfcReader.cc
+++ b/pn8x/halimpl/mifare/NxpMfcReader.cc
@@ -349,6 +349,10 @@ NFCSTATUS NxpMfcReader::AnalyzeMfcResp(uint8_t* pBuff, uint16_t* pBufflen) {
       } break;
 
       case eMfcAuthRsp: {
+        if (*pBufflen < 2) {
+          status = NFCSTATUS_FAILED;
+          break;
+        }
         /* check the status byte */
         if (NFCSTATUS_SUCCESS == pBuff[1]) {
           status = NFCSTATUS_SUCCESS;
diff --git a/snxxx/halimpl/mifare/NxpMfcReader.cc b/snxxx/halimpl/mifare/NxpMfcReader.cc
index 83fe1d9..7ac930b 100644
--- a/snxxx/halimpl/mifare/NxpMfcReader.cc
+++ b/snxxx/halimpl/mifare/NxpMfcReader.cc
@@ -353,6 +353,10 @@ NFCSTATUS NxpMfcReader::AnalyzeMfcResp(uint8_t* pBuff, uint16_t* pBufflen) {
       } break;
 
       case eMfcAuthRsp: {
+        if (*pBufflen < 2) {
+          status = NFCSTATUS_FAILED;
+          break;
+        }
         /* check the status byte */
         if (NFCSTATUS_SUCCESS == pBuff[1]) {
           status = NFCSTATUS_SUCCESS;
-- 
2.40.0.rc1.284.g88254d51c5-goog

